Makefile

```diff
@@ -0,0 +1,167 @@
+	# build those first by calling the templates rule.
+	cd $(MANUALDIR)
+	./format_files_for_userguide.py 
+	lualatex $(MANUALTEX)
+	lualatex $(MANUALTEX)
+	cd ..
+
+
+version:
+	# Upate version information and date of all moderncv files. A call make version
+	# will define VERSIONNEXT=$(shell git describe --tags). Alternatively, call
+	# "make version NEW=v5.13.298" to set v5.13.298 as the new version number.
+	# The date gets calculated by the date function.
+	if [[ $(strip $(VERSION)) == $(strip $(VERSIONNEXT)) ]]; then
+		echo "Old version number $(VERSION)  same as new version number $(VERSIONNEXT)"
+		echo "nothing to do, aborting."
+	else
+		# we need to split the $(VERSION) date format into substrings to using
+		# sed. This is due to / being a special character in sed.
+		YEAR=$(shell echo $(VERSIONDATE) | cut -d "/" -f 1)
+		MONTH=$(shell echo $(VERSIONDATE) | cut -d "/" -f 2)
+		DAY=$(shell echo $(VERSIONDATE) | cut -d "/" -f 3)
+		# update version info and date of *.sty, *.cls and *.tex files
+		# prepare find and replace with sed
+		findstr="$$YEAR\\/$$MONTH\\/$$DAY $(VERSION)"
+		replacestr="$(VERSIONDATENEXT) $(VERSIONNEXT)"
+		for currentdir in $(MODERNCVDIR) $(MANUALDIR); do
+			for file in $$currentdir/*.cls $$currentdir/*.sty $$currentdir/*.tex; do
+				if [[ -f "$$file" ]] && [[ ! -h "$$file" ]]; then
+					echo "updating version info of file $$file to $(VERSIONNEXT) (was $(VERSION))";
+					sed -i "s/$$findstr/$$replacestr/g" $$file;
+					# update version info in the title of documentation
+					sed -i "s/Package version $(VERSION)}/Package version $(VERSIONNEXT)}/g" $$file;
+				fi
+			done
+		done
+		unset findstr; unset replacestr
+		# update VERSION and VERSIONDATE variable of this very file
+		sed -i "s/VERSION = $(VERSION)/VERSION = $(VERSIONNEXT)/g" $(MODERNCVDIR)/Makefile
+		findstr="VERSIONDATE = $$YEAR\\/$$MONTH\\/$$DAY"
+		replacestr="VERSIONDATE = $(shell date +"%Y")\\/$(shell date +"%m")\\/$(shell date +"%d")"
+		sed -i "s/$$findstr/$$replacestr/g" $(MODERNCVDIR)/Makefile
+		unset findstr; unset replacestr
+	fi
+
+tarball:
+	# build a tarball for release puposes. If the examples directory exist,
+	# include them
+	# remove existing tarballs
+	rm -f *.gz *.tar
+	# create tar with all files in git repo
+	git archive HEAD > $(TARBALL)
+	# remove git specific files
+	tar -f $(TARBALL) --delete .github/ .gitignore create-release-tarball.sh
+	# if examples exist include them in the tarball
+	if [[ -d "$(EXAMPLESDIR)" ]]; then
+		tar -rf $(TARBALL) --append $(EXAMPLESDIR)
+	fi
+	# include precompiled template pdfs and userguide from manual folder,
+	# the idea being that the userguide can be build from the manual folder
+	# and has everything that it needs to compile. If the release method gets
+	# called this ensures that a precompiled version of the userguide is
+	# included in the tar ball.
+	tar -rf $(TARBALL) --append $(MANUALDIR)
+	# compress
+	gzip $(TARBALL)
```

Comment:
Ok first thing, with this you want to remove `create-release-tarball.sh` from the repo, as it is essentially the same. Second, the reason for the tarball is the upload to CTAN, so the example creating should be consistent. I personally don't think they should be included in the release tarball, only the manual.

The part where you add the manual should also be in the commit where the actual tarball will be added.

Reply:
I thought it might be a good thing to have examples precompiled for users to have a look at. Therefore I included them. 
To not clutter the repo with precompiled pdfs, I only included them in the release tar ball, so that they are also available on CTAN for download immediatelly.

Reply:
Yes I agree that precompiled pdf shouldn't be in the repo. I checked a couple of other packages, and I guess it's fine if we add one example to the release tarball which can be linked to at the CTAN page (I wouldn't add all of them, I think that's a bit useless).

In any case, it's important that this is consistent, and not depending on the existence of a folder.